flycheck.el

```diff
@@ -6079,7 +6084,17 @@ See URL `http://handlebarsjs.com/'."
           "Error: Parse error on line " line ":" (optional "\r") "\n"
           (zero-or-more not-newline) "\n" (zero-or-more not-newline) "\n"
           (message) line-end))
-  :modes (handlebars-mode handlebars-sgml-mode))
+  :modes (handlebars-mode handlebars-sgml-mode web-mode)
+  :predicate (lambda ()
+               (or (and (eq major-mode 'web-mode)
+                        (string-equal web-mode-engine "ctemplate")
+                        (string-equal "handlebars"
+                                      (catch 'done
+                                        (dolist (elt web-mode-engine-file-regexps)
+                                          (when (string-match-p (cdr elt) (buffer-file-name))
+                                            (throw 'done (car elt)))))))
```

Comment:
What does this `dolist` do, and what purpose does it serve?  It looks to me as if you have duplicated some logic from Web Mode itself…